game-state-service

```diff
@@ -24,12 +32,25 @@ class GameState(object):
         try:
             parser.load_from_file(self._FILE_PATH)
             root = parser.get_root()
-            assert root is not None
+            if root is None:
+                raise InvalidStateError()
             self._state = root.get_object()
-            assert self._state is not None
-        except (GLib.Error, AssertionError):
-            traceback.print_exc()
+            if self._state is None:
+                raise InvalidStateError()
+            reset = False
+        except GLib.Error as e:
+            if GLib.quark_from_string(e.domain) == GLib.file_error_quark():
+                _logger.error('Can\'t read from file %s' % self._FILE_PATH)
```

Comment:
I thought of one more thing: in the logger messages derived from `GLib.Error`s, append `e.message` to the end of the message.